spock-spring/src/main/java/org/spockframework/spring/xml/MockBeanDefinitionParser.java

```diff
@@ -28,10 +29,18 @@ protected String getBeanClassName(Element element) {
   @Override
   protected void doParse(Element element, BeanDefinitionBuilder builder) {
 
+    if (element.getLocalName().equals("spy") && element.getAttribute("ref").isEmpty()
+          && element.getAttribute("class").isEmpty()) {
+      throw new BeanCreationException(element.getAttribute("id"),
+             "Cannot create spy. Please specify either 'class' or 'ref' attributes.");
+    }
     // configure MockFactory
     builder.addConstructorArgValue(element.getAttribute("class"));
     builder.addPropertyValue("name", element.getAttribute("id"));
     builder.addPropertyValue("mockNature", element.getLocalName());
+    if(!element.getAttribute("ref").isEmpty()) {
```

Comment:
Yes I think it should fail fast with a helpful error message. As for the xsd, you could add it to the description, unfortunately there is no `choice` equivalent in XML Schema for attributes, there seems to be some possibility to achieve this with Schematron or Schemapath, but I don't think it would improve the tool support https://www.w3.org/wiki/Attribute_mutex